tests/workflows/test_io_output.py

```
@@ -0,0 +1,73 @@
+
+        with _open_output(str(output_file)) as handle:
+            handle.write(b"test\n")
+
+        assert output_file.read_bytes() == b"test\n", (
+            "file content should match what was written"
+        )
+
+    def test_empty_path_exits(self) -> None:
+        """An empty output path raises SystemExit."""
+        with pytest.raises(SystemExit, match=""):
+            with _open_output(""):
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)